### ISSUE TITLE
414 Request URI too long

### ISSUE THREAD

**Issue by nipuna-perera** [mozzarellathicc/attcerts#6]
When I try to get the host file I get this. I basically get this for any curl I try. Am I on too old of firmware? My RG is on 2.17.5

```html
<html>
<head>
<title>414 Request-URI Too Long</title>
</head>
<body>
<h1>414 Request-URI Too Long</h1>      
</body>
</html>
```

Trying this on windows, I tried both terminal and cmd prompt. Same error. I am assuming this is coming from the RG itself.

**Comment by tehdango** [comment]
IDK for sure but that is a major version older than whats suggested. I used the one provided in the readme and it worked without issue.

**Comment by nipuna-perera** [comment]
Ok thank you! I will try upgrading.

**Comment by EvanCarroll** [comment]
My upgrade failed. But I am also getting the same thing. I'm using version 2.17.5.

```
❯ curl --ignore-content-length -X"GET a/etc/hosts" http://192.168.1.254:80
<html>
<head>
<title>414 Request-URI Too Long</title>
</head>
<body>
<h1>414 Request-URI Too Long</h1>
</body>
</html>
```

**Comment by tehdango** [comment]
Use the version in the instructions and see if that works. That's the version I used and I had no issue.

**Comment by EvanCarroll** [comment]
@tehdango  I used the one in the instructions, you can't upgrade 3.18.2.

> http://gateway.c01.sbcglobal.net/firmware/ALPHA/210/001E46/BGW210-700_3.18.2/spTurquoise210-700_3.18.2_ENG.bin

**Comment by tehdango** [comment]
This is the latest firmware. Maybe it will let you upgrade then downgrade.

http://gateway.c01.sbcglobal.net/firmware/GA/210/001E46/BGW210-700_4.26.11/spTurquoise210-700_4.26.11.bin

**Comment by flytiger** [comment]
> The only firmware that worked for me to upgrade from 2.17.5 was the NO_AT one,
> 
> * http://gateway.c01.sbcglobal.net/firmware/ALPHA/210/001E46/BGW210-700_debug/spTurquoise210-700_2.14.2_NO_AT.bin
> 
> However, that firmware doesn't get me closer to 3.18.2, when I try to upgrade to 3.18.2_ENG I get,
> 
> > Firmware image is invalid.

I received a bgw210 with 2.17.5 firmware as well, the way I resolve the issue is to downgrade to 2.14.2.NO_AT then downgrade again to 2.7.7. This version has the same vulnerability so you can use the script to extract the certificates and key. Please note 2 points, 1, setup static IP on your client. This is **REQUIRED** for the downloading the mfg.dat. 2, wait for more than 2 mins, after plug in the power of the RG, the full mfg.dat file will be downloaded around that time. Hope this helps

**Comment by EvanCarroll** [comment]
Yes I think that's close to what I did except this whole repo didn't work for me. I got to 2.7.7 and then I went and did the https://github.com/iwleonards/extract-mfg method. The only catch there was that I had to have the `.der` files in the same directory as the `mfg.dat`  before I ran `mfg_dat_decode`.
